instaloader/structures.py

```diff
@@ -569,8 +569,8 @@ def _asdict(self):
     def _obtain_metadata(self):
         try:
             if not self._has_full_metadata:
-                metadata = self._context.get_json('{}/'.format(self.username), params={})
-                self._node = metadata['entry_data']['ProfilePage'][0]['graphql']['user']
+                metadata = self._context.get_json('{}/'.format(self.username), params={'__a': 1})
+                self._node = metadata['graphql']['user']
                 self._has_full_metadata = True
                 self._rhx_gis = metadata.get('rhx_gis')
```

Comment:
`self._rhx_gis` is needed for several queries. With your change, `rhx_gis` is not in `metadata` anymore, leading `self._rhx_gis` to be set to `None`. Do `Profile.get_*_posts()` still work after this change?

Reply:
@aandergr Profile.get_*_posts() is still working. I'm not familiar with what "rhx_gis" is required for, as it looks like Instagram has deprecated it.  If it is required for some endpoints which I'm not aware of, we should request to this endpoint twice, with and without "__a": 1 parameter, and check which result contains required data.